components/region_cache_memory_engine/src/background.rs

```diff
@@ -263,6 +283,59 @@ impl BgWorkManager {
 struct BackgroundRunnerCore {
     engine: Arc<RwLock<RangeCacheMemoryEngineCore>>,
     memory_controller: Arc<MemoryController>,
+    range_stats_manager: Option<RangeStatsManager>,
+}
+
+#[derive(Clone)]
+struct RangeStatsManager {
+    num_regions: usize,
+    info_provider: Arc<dyn RegionInfoProvider>,
+    prev_top_regions: Arc<Mutex<BTreeMap<u64, Region>>>,
+}
+
+impl RangeStatsManager {
+    pub fn new(num_regions: usize, info_provider: Arc<dyn RegionInfoProvider>) -> Self {
+        RangeStatsManager {
+            num_regions,
+            info_provider,
+            prev_top_regions: Arc::new(Mutex::new(BTreeMap::new())),
+        }
+    }
+
+    pub fn collect_changed_ranges(
+        &self,
+        ranges_added_out: &mut Vec<CacheRange>,
+        ranges_removed_out: &mut Vec<CacheRange>,
+    ) {
+        let curr_top_regions = self
+            .info_provider
+            .get_top_regions(self.num_regions)
```

Comment:
So we will deal with region label inside get_top_regions right?

Reply:
the region labels run independently: we will add top regions whenever they are added in a separate thread. however, once a labeled region is no longer one of the top, it will get evicted.